### PR TITLE
Fix under approximation of Make_block(Naked floats, _)

### PR DESCRIPTION
**Pull request by Gbury**
Should be merged only after #395

### CODE REVIEW

File: middle_end/flambda/types/structures/row_like.rec.mli [hunk 1]
```diff
@@ -22,6 +22,9 @@
    functor applications produce "undefined recursive module" errors. *)
 
 (* CR mshinwell: tidy up! *)
+(* CR gbury: Consider using Tag.Scannable.t instead of Tag.t in
+             For_blocks, and use another module to specifically handle float
+             arrays. *)
```
Comment (mshinwell) [inline middle_end/flambda/types/structures/row_like.rec.mli]: All-float blocks (e.g. records all of whose elements are `float`s), not float arrays, which are already handled separately.

File: middle_end/flambda/simplify/simplify_variadic_primitive.ml [hunk 2]
```diff
@@ -72,6 +72,51 @@ let simplify_make_block_of_values dacc _prim dbg tag ~shape
     Simplified_named.reachable term, env_extension, args, dacc
   end
 
+let simplify_make_block_of_floats dacc _prim dbg
+      ~(mutable_or_immutable : Mutability.t)
+      args_with_tys ~result_var =
+  let denv = DA.denv dacc in
+  let args, _arg_tys = List.split args_with_tys in
```
Comment (mshinwell) [inline middle_end/flambda/simplify/simplify_variadic_primitive.ml]: There doesn't seem any point in allocating `_arg_tys`, I suggest using `List.map fst args_with_tys` instead.

File: middle_end/flambda/simplify/simplify_variadic_primitive.ml [hunk 3]
```diff
@@ -72,6 +72,51 @@ let simplify_make_block_of_values dacc _prim dbg tag ~shape
...
+  let invalid () =
+    let ty = T.bottom K.value in
+    let env_extension = TEE.one_equation (Name.var result_var) ty in
+    Simplified_named.invalid (), env_extension, args, dacc
+  in
+  (* CR mshinwell: This could probably be done more neatly. *)
+  let found_bottom = ref false in
+  let fields =
+    List.map (fun ((arg : Simple.t), arg_ty) ->
+        if T.is_bottom (DE.typing_env denv) arg_ty then begin
```
Comment (mshinwell) [inline middle_end/flambda/simplify/simplify_variadic_primitive.ml:92]: Please add a CR comment here that we should review all similar pieces of code in the file and aim to remove the `T.is_bottom` checks (kind of like #336 did in the simplifier).

**Review by mshinwell — CHANGES_REQUESTED** [review]

File: middle_end/flambda/simplify/simplify_variadic_primitive.ml (hunk 3 above)
Comment (Gbury) [reply middle_end/flambda/simplify/simplify_variadic_primitive.ml:92]: done

File: middle_end/flambda/simplify/simplify_variadic_primitive.ml (hunk 2 above)
Comment (Gbury) [reply middle_end/flambda/simplify/simplify_variadic_primitive.ml]: done

File: middle_end/flambda/types/structures/row_like.rec.mli (hunk 1 above)
Comment (Gbury) [reply middle_end/flambda/types/structures/row_like.rec.mli]: fixed

**Comment by Gbury** [comment]
All comments have been addresed.

**Review by mshinwell — APPROVED** [review]

**Pull request merged**